docs/src/main/asciidoc/kotlin.adoc

```diff
@@ -173,6 +173,203 @@ for example, then `<option>all-open:annotation=javax.enterprise.context.Applicat
+    implementation 'io.quarkus:quarkus-kotlin'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured:{rest-assured-version}'  // <4>
+}
+
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
+
+test {
+    useJUnitPlatform()
+    exclude '**/Native*'
+}
+
+buildNative {
+    enableHttpUrlHandler = true
+}
+
+allOpen { // <5>
+    annotation("javax.ws.rs.Path")
+    annotation("javax.enterprise.context.ApplicationScoped")
+    annotation("io.quarkus.test.junit.QuarkusTest")
+}
+----
+
+<1> The Quarkus plugin needs to be applied.
+<2> The Kotlin plugin version needs to be specified.
+<3> We include the Quarkus BOM using Gradle's link:https://docs.gradle.org/5.4.1/userguide/managing_transitive_dependencies.html#sec:bom_import[relevant syntax]
+<4> The rest-assured version needs to be specified
+<5> All open configuration required, as per Maven guide above
+
+or, if you use the Gradle Kotlin DSL:
+
+[source,kotlin,subs=attributes+]
+----
+import io.quarkus.gradle.tasks.QuarkusNative
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "..."
+version = "1.0.0-SNAPSHOT"
+
+plugins {
+    java
+    id("io.quarkus") version "{quarkus-version}" // <1>
+
+    kotlin("jvm") version "{kotlin-version}" // <2>
+    id("org.jetbrains.kotlin.plugin.allopen") version "{kotlin-version}" // <2>
+}
+
+apply(plugin = "io.quarkus")
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation(kotlin("stdlib-jdk8"))
+
+    // <3>
+    implementation(enforcedPlatform("io.quarkus:quarkus-bom:${quarkusVersion}"))
+    implementation("io.quarkus:quarkus-kotlin")
+    implementation("io.quarkus:quarkus-resteasy")
+    implementation("io.quarkus:quarkus-resteasy-jsonb")
+
+    testImplementation("io.quarkus:quarkus-junit5")
+    testImplementation("io.rest-assured:rest-assured:{rest-assured-version}") // <4>
+}
+
+tasks {
+    named<QuarkusNative>("buildNative") {
+        setEnableHttpUrlHandler(true)
+    }
+}
+
+tasks.test {
+    useJUnitPlatform()
+    exclude '**/Native*'
+}
+
+allOpen { // <5>
+    annotation("javax.ws.rs.Path")
+    annotation("javax.enterprise.context.ApplicationScoped")
+    annotation("io.quarkus.test.junit.QuarkusTest")
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+val compileKotlin: KotlinCompile by tasks
+compileKotlin.kotlinOptions {
+    jvmTarget = "1.8"
+}
+
+val compileTestKotlin: KotlinCompile by tasks
+compileTestKotlin.kotlinOptions {
+    jvmTarget = "1.8"
+}
+----
+
+<1> The Quarkus plugin needs to be applied.
+<2> The Kotlin plugin version needs to be specified.
+<3> We include the Quarkus BOM using Gradle's link:https://docs.gradle.org/5.4.1/userguide/managing_transitive_dependencies.html#sec:bom_import[relevant syntax]
+<4> The rest-assured version needs to be specified
+<5> All open configuration required, as per Maven guide above
+
+== CDI @Inject with Kotlin
```

Comment:
Great idea to include this section!

Reply:
Issues with CDI & Kotlin initially put me off using Quarkus, it took a bit of experimenting and StackOverflow to solve it.  I thought it might save someone else the pain, if it's explicitly mentioned.  However I do note that the Kotlin CDI support has improved in the current release (the same @field injection failure is no longer an issue).